internal/kvprocessor/kvprocessor.go

```diff
@@ -52,8 +52,8 @@ func NewKVProcessor(
 }
 
 // ProcessAndSubmit processes relay.MessageKV. The main method which does all the work of the KVProcessor
-func (p *KVProcessor) ProcessAndSubmit(ctx context.Context, m *relay.MessageKV) error {
-	latestHeight, err := p.targetChain.ChainProvider.QueryLatestHeight(ctx)
+func (p *KVProcessor) ProcessAndSubmit(m *relay.MessageKV) error {
```

Comment:
why do we use background here?

Reply:
because this method is called by the `Relayer` during a KV query result submission, and we want to finish this process when an app termination occurs. if we passed the context from the `Relayer` here, it would lead the `QueryLatestHeight` exited immediately on termination with a context deadline error, e.g. no waiting for the submission to be finished. this `context.Background` passing the `QueryLatestHeight` stop buck to the `p.targetChain.ChainProvider` which can only be interrupted because of a timeout

Reply:
refactored. please give it another look